runtime/compiler/control/JITaaSCompilationThread.hpp

```diff
@@ -295,19 +295,71 @@ class CompilationInfoPerThreadRemote : public TR::CompilationInfoPerThread
       bool getWaitToBeNotified() const { return _waitToBeNotified; }
       void setWaitToBeNotified(bool b) { _waitToBeNotified = b; }
 
-      void clearIProfilerMap();
       bool cacheIProfilerInfo(TR_OpaqueMethodBlock *method, uint32_t byteCodeIndex, TR_IPBytecodeHashTableEntry *entry);
       TR_IPBytecodeHashTableEntry *getCachedIProfilerInfo(TR_OpaqueMethodBlock *method, uint32_t byteCodeIndex, bool *methodInfoPresent);
       void cacheResolvedMethod(TR_ResolvedMethodKey key, TR_OpaqueMethodBlock *method, uint32_t vTableSlot, TR_ResolvedJ9JITaaSServerMethodInfo &methodInfo);
       bool getCachedResolvedMethod(TR_ResolvedMethodKey key, TR_ResolvedJ9JITaaSServerMethod *owningMethod, TR_ResolvedMethod **resolvedMethod, bool *unresolvedInCP = NULL);
       TR_ResolvedMethodKey getResolvedMethodKey(TR_ResolvedMethodType type, TR_OpaqueClassBlock *ramClass, int32_t cpIndex, TR_OpaqueClassBlock *classObject=NULL);
-      void clearResolvedMethodInfoMap(TR_Memory *trMemory);
 
-      void clearResolvedMirrorMethodsPersistIPInfo();
       void cacheResolvedMirrorMethodsPersistIPInfo(TR_ResolvedJ9Method *resolvedMethod);
       ResolvedMirrorMethodsPersistIP_t *getCachedResolvedMirrorMethodsPersistIPInfo() { return _resolvedMirrorMethodsPersistIPInfo; }
 
+      void clearPerCompilationCaches();
+
    private:
+      /* Template method for allocating a cache of type T on the heap.
+       * Cache pointer must be NULL.
+       */
+      template <typename T>
+      bool initializePerCompilationCache(T* &cache)
+         {
+         // Initialize map
+         TR_ASSERT(!cache, "Cache already initialized");
+         TR_Memory *trMemory = getCompilation()->trMemory();
+         cache = new (trMemory->trHeapMemory()) T(typename T::allocator_type(trMemory->heapMemoryRegion()));
+         return cache != NULL;
+         }
+      /* Template method for storing key-value pairs (of types K and V respectively)
+       * to a heap-allocated unordered map.
+       * If a map is NULL, will allocate it.
+       */
+      template <typename K, typename V>
+      void cacheToPerCompilationMap(UnorderedMap<K, V>* &map, const K &key, const V &value)
+         {
+         if (!map) initializePerCompilationCache(map);
```

Comment:
Just to follow our coding convention, please move`initializePerCompilationCache(map);` on the next line